cmip6_downscaling/config/config.py

```diff
@@ -1,104 +1,130 @@
+            "EXTRA_PIP_PACKAGES": self.extra_pip_packages,
+            "OMP_NUM_THREADS": self.OMP_NUM_THREADS,
+            "MPI_NUM_THREADS": self.MPI_NUM_THREADS,
+            "MKL_NUM_THREADS": self.MKL_NUM_THREADS,
+            "OPENBLAS_NUM_THREADS": self.OPENBLAS_NUM_THREADS,
+            "DASK_DISTRIBUTED__WORKER__RESOURCES__TASKSLOTS": self.dask_distributed_worker_resources_taskslots,
+        }
+        return env
+
+    def dask_executor(self):
+
+        pod_spec = make_pod_spec(
+            image=self.image,
+            memory_limit=self.pod_memory_limit,
+            memory_request=self.pod_memory_request,
+            threads_per_worker=self.pod_threads_per_worker,
+            cpu_limit=self.pod_cpu_limit,
+            cpu_request=self.pod_cpu_request,
+            env=self.generate_env(),
+        )
+
+        pod_spec.spec.containers[0].args.extend(["--resources", "TASKSLOTS=1"])
+
+        daskExecutor = DaskExecutor(
+            cluster_class=lambda: KubeCluster(
+                pod_spec,
+                deploy_mode=self.deploy_mode,
+                adapt_kwargs={"minimum": self.adapt_min, "maximum": self.adapt_max},
             )
-        ),
-        adapt_kwargs={
-            "minimum": return_dask_config()["dask_executor_adapt_min"],
-            "maximum": return_dask_config()["dask_executor_adapt_max"],
-        },
-    )
-    return dask_executor
+        )
+        return daskExecutor
+
+    def kubernetes_run_config(self):
+        kube_run_config = KubernetesRun(
+            cpu_request=self.kubernetes_cpu,
+            memory_request=self.kubernetes_memory,
+            image=self.image,
+            labels=self.agent,
+            env=self.generate_env(),
+        )
+        return kube_run_config
+
+
+class PangeoConfig(AbstractConfig):
+    # with Flow(name="bcsd-testing", storage=storage, run_config=run_config) as flow:
+    # Which run_config? Does this need a dask executor?
+    def __init__(self):
+        super().__init__()
+
+
+class LocalConfig(AbstractConfig):
+    # what additional args to add?
+    def __init__(self):
+        super().__init__()
+
+    def local_run_config(self):
+        local_run = LocalRun()
+        return local_run
+
+
+class TestConfig(LocalConfig):
+    # for use in GitHub Actions
+    # what additional args to add for testconfig? additional pip packages?
+    def __init__(self):
+        super().__init__()
+
+    def local_run_config(self):
+        local_run = LocalRun(env={"EXTRA_PIP_PACKAGES": self.extra_pip_packages})
+        return local_run
+
+
+def get_config(**kwargs):
+
+    if os.environ.get("CI") == "true":
+        config = LocalConfig(**kwargs)
+    elif os.environ.get("PREFECT__BACKEND") == "cloud":
+        config = CloudConfig(**kwargs)
+    elif os.environ.get("PANGEO__BACKEND") == "pangeo":
+        config = PangeoConfig(**kwargs)
+    elif os.environ.get("TEST") == "test":
+        config = TestConfig(**kwargs)
```

Comment:
needs and `else` with a `ValueError` to catch when none of the expected environment variables are set.